delta/plugins/search/src/test/scala/ch/epfl/bluebrain/nexus/delta/plugins/search/SearchConfigUpdateFunctionSuite.scala

```diff
@@ -0,0 +1,131 @@
+import ch.epfl.bluebrain.nexus.testkit.postgres.Doobie
+import io.circe.Json
+import munit.AnyFixture
+
+import java.time.Instant
+import scala.concurrent.duration.DurationInt
+
+class SearchConfigUpdateFunctionSuite extends BioSuite with CompositeViewsFixture with Fixtures with Doobie.Fixture {
+  override def munitFixtures: Seq[AnyFixture[_]] = List(doobie)
+  private lazy val xas                           = doobie()
+
+  implicit private val s: Identity.User = subject
+  implicit private val baseUri: BaseUri = BaseUri("http://localhost", Label.unsafe("v1"))
+
+  private val fetchContext = FetchContextDummy[CompositeViewRejection](
+    Map(project.ref -> project.context),
+    Set.empty[ProjectRef],
+    ProjectContextRejection
+  )
+
+  private lazy val compositeViews = CompositeViews(
+    fetchContext,
+    ResolverContextResolution(rcr),
+    alwaysValidate,
+    crypto,
+    config,
+    xas
+  )
+
+  private val viewDef = ActiveViewDef(ViewRef(projectRef, id), uuid, 1, viewValue)
+
+  private val newProjections: NonEmptySet[CompositeViewProjectionFields] =
+    NonEmptySet.of(esProjectionFields)
+  private val newSources: NonEmptySet[CompositeViewSourceFields]         =
+    NonEmptySet.of(projectFields)
+  private val newRebuildStrategy: Option[RebuildStrategy]                =
+    Some(Interval(2.minutes))
+  private val newViewFields                                              =
+    viewFields.copy(
+      projections = newProjections,
+      sources = newSources,
+      rebuildStrategy = newRebuildStrategy
+    )
+
+  /** Create a view using [[CompositeViewsFixture]] defaults and check it is correctly created */
+  private def viewCreated(views: CompositeViews) =
+    for {
+      created <- views.create(id, projectRef, viewFields)
+      _        = assertEquals(
+                   created.copy(value = created.value.copy(source = Json.obj())),
+                   resourceFor(id, viewValue, source = Json.obj())
+                 )
+    } yield ()
+
+  /**
+    * Using the SearchConfigUpdater update function update the given [[ActiveViewDef]] using the new
+    * [[CompositeViewFields]]
+    */
+  private def updateView(views: CompositeViews, viewDef: ActiveViewDef, newFields: CompositeViewFields) = {
+    val update = SearchConfigUpdater.update(views)
+    update(viewDef, newFields)
+  }
+
+  test("Sources, projections, and rebuild strategy should be updated") {
+    for {
+      views   <- compositeViews
+      _       <- viewCreated(views)
+      _       <- updateView(views, viewDef, newViewFields)
+      updated <- views.fetch(id, projectRef)
+      _        = assertEquals(updated.rev, 2)
+      _        = assertEquals(updated.value.sources, newSources.map(_.toSource(uuid, id)))
+      _        = assertEquals(updated.value.projections, newProjections.map(_.toProjection(uuid, id)))
+      _        = assertEquals(updated.value.rebuildStrategy, newRebuildStrategy)
+    } yield ()
+  }
+
```

Comment:
This is only a suggestion, but this is a little hard to follow because some of the things are happening implicitly. Consider this alternative:

```
for {
      views   <- thereIsAView(viewRef, viewFields)
      _       <- updateView(views, viewRef, newViewFields)
      updated <- views.fetch(viewRef)
      _        = assertEquals(updated.rev, 2)
      _        = assertEquals(updated.value.sources, newSources.map(_.toSource(uuid, id)))
      _        = assertEquals(updated.value.projections, newProjections.map(_.toProjection(uuid, id)))
      _        = assertEquals(updated.value.rebuildStrategy, newRebuildStrategy)
    } yield ()
  }
```

Reply:
The main change here is to be explicit about the view you are creating/updating/reading, it makes reading clearer even though you don't need to pass the values in to make the function work

Similar to this, if the new fields were more obviously about the sources, projections and strategy then that would make it easier to see why the assertions are happening

I'm only posting this to show my perspective, the test is not a problem